packages/repository/src/entity/interface/data.ts

```diff
@@ -1,14 +1,10 @@
+import { Simplify } from "type-fest"
+
 export type UserDefinedSchema = Record<string, any>
 
 export type EntitySchema<TInput extends UserDefinedSchema = UserDefinedSchema> =
-  TInput["id"] extends unknown
-    ? Omit<TInput, "id"> & { id: string }
-    : TInput & { id: string }
+  Simplify<Omit<TInput, "id"> & { id: string }>
```

Comment:
Mam pytanie
`<TInput extends UserDefinedSchema = UserDefinedSchema>`

Rozumiem, że ten zapis powoduje, że `TInput` exptenduje z `UserDefinedSchema` ale po co jeszcze na końcu jest to `= UserDefinedSchema`?

Reply:
Jasne, chodzi o że `TInput` musi być jak `UserDefinedSchema` (tego typu lub bardziej dokładny, jest na to słowo - kowariancja, więcej tutaj: https://learn.microsoft.com/pl-pl/dotnet/standard/generics/covariance-and-contravariance). Za to zachowanie odpowiada `extends`. Natomiast `=` przypisuje domyślną wartość dla tego generyka - jesli jej nie podamy `TInput` będzie równy `UserDefinedSchema`.

Taki zabieg jest przydatny, jeśli generalnie typ ma nieść informacje pochodzące z inferencji (np. przyjmować typ wprowadzonego argumentu podczas wywołania funkcji) ale w innych miejscach zamiast drillować przekazując wartość generyka, można podłożyć ogólny schemat.